database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,597 @@
+        chart_data.update_every = sqlite3_column_int(res, 7);
+        chart_data.chart_type = sqlite3_column_int(res, 8);
+        chart_data.family = (char *) sqlite3_column_text(res, 9);
+        dict_cb(&chart_data, data);
+    }
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement that fetches chart label data, rc = %d", rc);
+}
+
+// Dimension list
+#define CTX_GET_DIMENSION_LIST  "SELECT d.dim_id, d.id, d.name FROM meta.dimension d WHERE d.chart_id = @id;"
+void ctx_get_dimension_list(uuid_t *chart_uuid, void (*dict_cb)(SQL_DIMENSION_DATA *, void *), void *data)
+{
+    int rc;
+    sqlite3_stmt *res = NULL;
+
+    rc = sqlite3_prepare_v2(db_context_meta, CTX_GET_DIMENSION_LIST, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to fetch chart dimension data");
+        return;
+    }
+
+    rc = sqlite3_bind_blob(res, 1, *chart_uuid, sizeof(*chart_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind chart_id to fetch dimension list");
+        goto failed;
+    }
+
+    SQL_DIMENSION_DATA dimension_data;
+
+    while (sqlite3_step(res) == SQLITE_ROW) {
+        uuid_copy(dimension_data.dim_id, *((uuid_t *)sqlite3_column_blob(res, 0)));
+        dimension_data.id = (char *) sqlite3_column_text(res, 1);
+        dimension_data.name = (char *) sqlite3_column_text(res, 2);
+        dict_cb(&dimension_data, data);
+    }
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement that fetches the chart dimension list, rc = %d", rc);
+}
+
+// LABEL LIST
+#define CTX_GET_LABEL_LIST  "SELECT l.label_key, l.label_value, l.source_type FROM meta.chart_label l WHERE l.chart_id = @id;"
+void ctx_get_label_list(uuid_t *chart_uuid, void (*dict_cb)(SQL_CLABEL_DATA *, void *), void *data)
+{
+    int rc;
+    sqlite3_stmt *res = NULL;
+
+    rc = sqlite3_prepare_v2(db_context_meta, CTX_GET_LABEL_LIST, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to fetch chart lanbels");
+        return;
+    }
+
+    rc = sqlite3_bind_blob(res, 1, *chart_uuid, sizeof(*chart_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind chart_id to fetch chart labels");
+        goto failed;
+    }
+
+    SQL_CLABEL_DATA label_data;
+
+    while (sqlite3_step(res) == SQLITE_ROW) {
+        label_data.label_key = (char *) sqlite3_column_text(res, 0);
+        label_data.label_value = (char *) sqlite3_column_text(res, 1);
+        label_data.label_source = sqlite3_column_int(res, 2);
+        dict_cb(&label_data, data);
+    }
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement that fetches chart label data, rc = %d", rc);
+
+    return;
```

Comment:
You did not add `return` in the previous functions, so to keep an unique pattern, please, remove it here.